SingularityService/src/main/java/com/hubspot/singularity/scheduler/SingularityUpstreamChecker.java

```diff
@@ -0,0 +1,80 @@
+package com.hubspot.singularity.scheduler;
+
+import java.util.Collection;
+import java.util.List;
+
+import javax.inject.Inject;
+import javax.inject.Singleton;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.common.base.Optional;
+import com.google.common.collect.Lists;
+import com.hubspot.baragon.models.UpstreamInfo;
+import com.hubspot.mesos.protos.MesosParameter;
+import com.hubspot.singularity.SingularityTask;
+import com.hubspot.singularity.data.TaskManager;
+import com.hubspot.singularity.helpers.MesosUtils;
+import com.hubspot.singularity.hooks.LoadBalancerClient;
+import com.hubspot.singularity.hooks.LoadBalancerClientImpl;
+
+@Singleton
+public class SingularityUpstreamChecker {
+
+  private static final Logger LOG = LoggerFactory.getLogger(LoadBalancerClient.class);
+
+  private final LoadBalancerClientImpl lbClient;
+  private final TaskManager taskManager;
+
+  @Inject
+  public SingularityUpstreamChecker(LoadBalancerClientImpl lbClient, TaskManager taskManager) {
+    this.lbClient = lbClient;
+    this.taskManager = taskManager;
+  }
+
+  private UpstreamInfo taskToUpstreamInfo (SingularityTask task, Optional<String> loadBalancerUpstreamGroup) {
```

Comment:
since this is pretty much copied from the `LoadBalancerClientImpl`, could be worth factoring it out into a public or static method we can call from there